feat: allow pydantic v2 usage

cfbevan opened the pull request:
* feat: start migration to pydantic v2
* update tests for pydantic v2

Followed https://docs.pydantic.dev/latest/migration/#using-pydantic-v1-features-in-a-v1v2-environment

Note that current documentation references a CONTRIBUTING.md that does not exist. The GitHub Pages site encourages reaching out for instructions on contributing, but no contact information is listed. Please feel free to comment on any changes needed.

cfbevan committed:
feat: allow pydantic v2 usage

* feat: start migration to pydantic v2
* update tests for pydantic v2

nadobando commented on pyproject.toml:
```diff
@@ -100,7 +100,7 @@ version = "0.2.1"
 aioarango = "^1.0.0"
 indexed = "^1.3.0"
-pydantic = "==1.10.12"
+pydantic = ">=1.10.17"
```
maybe we can already set this to 
```suggestion
pydantic = "<2"
```

cfbevan replied:
The documentation claims that the `v1` modules were not added till 1.10.17. Another option is to update the imports to something like this:

```python
try:
    from pydantic.v1.fields import ModelField
except ImportError:
    from pydantic.fields import ModelField 
```

nadobando reviewed (approved):
gr8 initiative @cfbevan

cfbevan commented:
I'm a bit confused by the CI checks. They ran locally fine. I will take a look at the errors and update as needed.

cfbevan commented:
I see my issue. The CI/CD uses 3.9 and I was using 3.11 locally.

cfbevan committed:
fix: fix ci/cd issues

- Fix issues raised in ci/cd for #15 that came from testing with python
  3.11 locally instead of python 3.9

cfbevan commented:
All tests passed. There is something wrong with the tokens.

nadobando commented:
`CI / Tests` is failing because of the token permissions, I'll will merge and open an issue :) 
thanks for you sharing
